crates/net/network/src/transactions.rs

```diff
@@ -808,10 +808,65 @@ mod tests {
             peer_id,
             msg: Transactions(vec![TransactionSigned::default()]),
         });
-
+        // this is a no-op assert that doesn't actually test the SyncState
+        // logic above, a rand PeerId short-circuits even if
+        // SyncState::Idle
         assert!(pool.is_empty());
     }
 
+    #[tokio::test(flavor = "multi_thread")]
+    #[cfg_attr(not(feature = "geth-tests"), ignore)]
+    async fn test_tx_broadcasts_through_two_syncs() {
+        reth_tracing::init_test_tracing();
+
+        let secret_key = SecretKey::new(&mut rand::thread_rng());
+
+        let client = NoopProvider::default();
+        let pool = testing_pool();
+        let config = NetworkConfigBuilder::new(secret_key)
+            .disable_discovery()
+            .listener_port(0)
+            .build(client);
+        let (handle, network, mut transactions, _) = NetworkManager::new(config)
+            .await
+            .unwrap()
+            .into_builder()
+            .transactions(pool.clone())
+            .split_with_handle();
+
+        tokio::task::spawn(network);
+
+        // go to syncing (pipline sync) to idle and then to syncing (live)
+        handle.update_sync_state(SyncState::Syncing);
+        assert!(NetworkInfo::is_syncing(&handle));
+        handle.update_sync_state(SyncState::Idle);
+        assert!(!NetworkInfo::is_syncing(&handle));
+        handle.update_sync_state(SyncState::Syncing);
+        assert!(NetworkInfo::is_syncing(&handle));
+
+        let peer_id = PeerId::random();
+        transactions.on_network_tx_event(NetworkTransactionEvent::IncomingTransactions {
+            peer_id,
+            msg: Transactions(vec![TransactionSigned::default()]),
+        });
+        assert!(!NetworkInfo::is_initially_syncing(&handle));
+        assert!(NetworkInfo::is_syncing(&handle));
+        // assert!(!pool.is_empty()); this keeps failing - indicating that there's a bug with these
+        // tests
+
+        // if we attempt to modify the initial test above with SyncState::Idle, the same assert
```

Comment:
cc @mattsse re these tests

tl;dr - `assert pool.is_empty()` and `assert !pool.is_empty()` don't seem to be doing what we want it to in these tests (?)

Reply:
what do we want here?

not following